### PR TITLE
ssl: introduce SSLEngineConfigurator#setSSLParameters

### PR DESCRIPTION
**Pull request by kofemann**
SSLParameters is a class introduced in java6 which can take over
SSLEngineConfigurator. As a first step, let allow applications to
use SSLParameters to configure SSLEngineConfigurator.

### CODE REVIEW

**Review by bshannon — APPROVED** [review]
I suppose this is fine, although it's interesting that SSLParameters supports a lot of
options that are ignored here, and SSLEngine can take an SSLParamaters object
directly, but that isn't done here.  I suspect this code should be updated to handle
additions to SSLEngine in JDK 1.7 and 1.8.

**Comment by kofemann** [comment]
As mentioned in the commit message, this can be the first step. The SSLEngineConfigurator can be updated to build SSLParamaters and then use it when SSLEngineConfigurator#configure method is called. This will make code much simple.

File: modules/grizzly/src/main/java/org/glassfish/grizzly/ssl/SSLEngineConfigurator.java [hunk 1]
```diff
@@ -272,6 +273,21 @@ public SSLEngineConfigurator setWantClientAuth(boolean wantClientAuth) {
         return this;
     }
 
+    /**
+     * Apply {@link SSLParameters} to this SSLEngineConfigurator.
```
Comment (arjantijms) [inline modules/grizzly/src/main/java/org/glassfish/grizzly/ssl/SSLEngineConfigurator.java:246]: > Apply {@link SSLParameters} to this SSLEngineConfigurator.

Might want to document that only the cipherSuites, protocols, needClientAuth and wantClientAuth properties are applied?

**Review by arjantijms — CHANGES_REQUESTED** [review]

**Comment by kofemann** [comment]
may be make sense to squash this two commits...

**Comment by mnriem** [comment]
@kofemann Have you applied the requested changes?

**Comment by kofemann** [comment]
Are people interested in this changes? In general, looks like grizzly project has very low activity since eclipse took over.

**Comment by arjantijms** [comment]
> Are people interested in this changes? In general, looks like grizzly project has very low activity since eclipse took over.

We definitely have a low amount of resources to work on Grizzly, which is really unfortunate :(

**Review by arjantijms — APPROVED** [review]
Let's forget about the requested changes

**Comment by kofemann** [comment]
Ok. Let me rebase and retest. I will update the PR.

**Comment by mnriem** [comment]
@arjantijms  Looks good to me

**Pull request merged**